### PR TITLE
Create streams info table based on processor type table

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *Create streams info table based on processor type table*
 - *Add test cases based on simulation data for all scenarios*
 - *Update description for streams info table, processor type table and CPU mapping table*

### Tickets:
 - *101248*

### CODE REVIEW

File: src/plugins/intel_cpu/src/plugin.h [hunk 1]
```diff
@@ -37,6 +37,11 @@ class Engine : public InferenceEngine::IInferencePlugin {
     InferenceEngine::IExecutableNetworkInternal::Ptr ImportNetwork(std::istream& networkModel,
                                                      const std::map<std::string, std::string>& config) override;
 
+    std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
+                                                         const int input_threads,
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/plugin.h:41]: I suppose it should not be a plugin method

Comment (wangleis) [reply src/plugins/intel_cpu/src/plugin.h:41]: This method is based on CPU abstract information and decoupled with HW platform. 

Putting this unified method in plugin give one more option which one CPU plugin work with different StreamsExecutor for different HW platforms.

**Comment by wangleis** [comment]
@dmitry-gorokhov @ilya-lavrenov Files are updated. Please review again.

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/src/cpu_streams_calculation.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include <vector>
+
+namespace ov {
+namespace intel_cpu {
+/**
+ * @brief      Generate streams information table according to processors type table
+ * @param[in]  input_streams is target streams set by user via NUM_STREAMS or hints.
+ *               - input "0" mean function generate the optimal number of streams
+ *               - LATENCY hint equals 1 stream.
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.hpp:20]: Is it true for multisocket systems? I assume there should be amount equal to number of Numa nodes.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**